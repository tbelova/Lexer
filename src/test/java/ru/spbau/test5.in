
read a; 

x := 0; 
y := 1; 

while (a >= 1) do { 
	x := x + 1; 
	y := x * x + 10 - 7 * (y + 1); 
	a := a - 1; 
} 

write x;
write y;
           
