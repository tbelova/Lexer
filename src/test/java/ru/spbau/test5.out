PROGRAM((2,0)(16,4))
   STATEMENTS((2,0)(14,7))
      STATEMENT((2,0)(2,6))
         READ_STATEMENT((2,0)(2,5))
            IDENT((2,5)(2,5))[a]
      STATEMENTS((4,0)(14,7))
         STATEMENT((4,0)(4,6))
            ASSIGNMENT((4,0)(4,5))
               IDENT((4,0)(4,0))[x]
               EXPRESSION((4,5)(4,5))
                  INTEGER((4,5)(4,5))[0]
         STATEMENTS((5,0)(14,7))
            STATEMENT((5,0)(5,6))
               ASSIGNMENT((5,0)(5,5))
                  IDENT((5,0)(5,0))[y]
                  EXPRESSION((5,5)(5,5))
                     INTEGER((5,5)(5,5))[1]
            STATEMENTS((7,0)(14,7))
               STATEMENT((7,0)(11,0))
                  WHILE_STATEMENT((7,0)(11,0))
                     BOOLEAN_EXPRESSION((7,7)(7,12))
                        GE((7,7)(7,12))
                           EXPRESSION((7,7)(7,7))
                              IDENT((7,7)(7,7))[a]
                           EXPRESSION((7,12)(7,12))
                              INTEGER((7,12)(7,12))[1]
                     BLOCK((7,18)(11,0))
                        STATEMENTS((8,1)(10,11))
                           STATEMENT((8,1)(8,11))
                              ASSIGNMENT((8,1)(8,10))
                                 IDENT((8,1)(8,1))[x]
                                 EXPRESSION((8,6)(8,10))
                                    SUM((8,6)(8,10))
                                       IDENT((8,6)(8,6))[x]
                                       INTEGER((8,10)(8,10))[1]
                           STATEMENTS((9,1)(10,11))
                              STATEMENT((9,1)(9,30))
                                 ASSIGNMENT((9,1)(9,29))
                                    IDENT((9,1)(9,1))[y]
                                    EXPRESSION((9,6)(9,29))
                                       SUB((9,6)(9,29))
                                          SUM((9,6)(9,15))
                                             MULT((9,6)(9,10))
                                                IDENT((9,6)(9,6))[x]
                                                IDENT((9,10)(9,10))[x]
                                             INTEGER((9,14)(9,15))[10]
                                          MULT((9,19)(9,29))
                                             INTEGER((9,19)(9,19))[7]
                                             SUM((9,24)(9,28))
                                                IDENT((9,24)(9,24))[y]
                                                INTEGER((9,28)(9,28))[1]
                              STATEMENTS((10,1)(10,11))
                                 STATEMENT((10,1)(10,11))
                                    ASSIGNMENT((10,1)(10,10))
                                       IDENT((10,1)(10,1))[a]
                                       EXPRESSION((10,6)(10,10))
                                          SUB((10,6)(10,10))
                                             IDENT((10,6)(10,6))[a]
                                             INTEGER((10,10)(10,10))[1]
               STATEMENTS((13,0)(14,7))
                  STATEMENT((13,0)(13,7))
                     WRITE_STATEMENT((13,0)(13,6))
                        EXPRESSION((13,6)(13,6))
                           IDENT((13,6)(13,6))[x]
                  STATEMENTS((14,0)(14,7))
                     STATEMENT((14,0)(14,7))
                        WRITE_STATEMENT((14,0)(14,6))
                           EXPRESSION((14,6)(14,6))
                              IDENT((14,6)(14,6))[y]
